core.js

```diff
@@ -736,7 +736,8 @@ async function _fromTokenizer(tokenizer) {
 			while (children > 0) {
 				const element = await readElement();
 				if (element.id === 0x42_82) {
-					return tokenizer.readToken(new Token.StringType(element.len, 'utf-8')); // Return DocType
+					const rawValue = await tokenizer.readToken(new Token.StringType(element.len, 'utf-8'));
+					return rawValue.replace(/\00.*$/g, ''); // Return DocType
```

Comment:
Does it document the maximum amount of null characters there could be? Would be nice to have a limit in place so it wouldn't hang on faulty files that has too many null characters.

Reply:
There is no maximum, it's used as a kind of padding. 
The maximum length is of the string read is already terminated by `element.len`.

Reply:
Ah you mean check `element.len`? It can be 2^35 (exceeding `number`) and the length is encoded a specific way ([VINT examples](https://tools.ietf.org/id/draft-ietf-cellar-ebml-03.html#vint-examples)). 
At that point the assumption is already it is EBML and starts to consume the tokenizer
and iterate through the [_EBML elements_](https://tools.ietf.org/id/draft-ietf-cellar-ebml-03.html#notation-and-conventions).
You could say, the docType must be a relative short value, but then we already matched `0x1A, 0x45, 0xDF, 0xA3` & `0x42_82`. Extremely unlikely we hit that point without the format being EBML.